### PR TITLE
Add CLI autoconfig

### PR DESCRIPTION
**Pull request by meffij**

### CODE REVIEW

File: api/python/tests/test_cli.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class CommandLineTestCase(TestCase):
+    """
+    Base TestCase class, sets up a CLI parser
+    """
+    @classmethod
+    def setUpClass(cls):
+        parser = create_parser()
+        cls.parser = parser
+
+class PingTestCase(CommandLineTestCase):
```
Comment (stevededalus) [inline api/python/tests/test_cli.py]: Maybe not named `PingTestCase` :-P

File: api/python/t4/main.py [hunk 2]
```diff
@@ -25,6 +25,12 @@ def create_parser():
     logout_p = subparsers.add_parser("logout", description=shorthelp, help=shorthelp)
     logout_p.set_defaults(func=session.logout)
 
+    shorthelp = "Configure T4"
+    config_p = subparsers.add_parser("config", description=shorthelp, help=shorthelp)
+    config_p.add_argument("catalog_url", help="URL of catalog to config with",
```
Comment (stevededalus) [inline api/python/t4/main.py]: Is there a public global catalog we can default to, or is that dead?

Comment (meffij) [reply api/python/t4/main.py]: I think we decided that wasn't too helpful -- we want people to go through the catalog for login anyway, so they should have a catalog they can config against.

Comment (stevededalus) [reply api/python/t4/main.py]: sgtm

**Pull request merged**